### PR TITLE
Task/register user api

### PR DESCRIPTION
**Pull request by marccolm**
#### What's this PR do?
Creates user registration API and tests, it adds djoser to the settings

#### Where should the reviewer start?
settings, users serializer, users urls and users tests

#### Any background context you want to provide?

This template was adapted ~stolen~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: users/test_views.py [hunk 1]
```diff
@@ -16,13 +16,69 @@ def setUp(self):
...
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_do_not_send_reset_email(self):
+        """Test that calling the resest password API returns a failed response if mail doesn't exist.
+        """
+        url = reverse('users:api_reset_password')
+        data = {'email': 'does_not_exist@gmail.com'}
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
```
Comment (fernandolobato) [inline users/test_views.py]: Is there a way to get the link the UI for the url for reseting the password and sending a POST request to it?

File: users/test_views.py [hunk 2]
```diff
@@ -16,13 +16,69 @@ def setUp(self):
...
+
+class UserRegistrationTest(APITestCase):
+    def test_create_user(self):
+        """Test that an object instance can be generated through the REST API endpoint.
+        """
+        url = reverse('users:api_registration')
+        data = {
+                'username': 'marcoantonio@gmail.com', 'password': 'marcoantonio123456',
```
Comment (fernandolobato) [inline users/test_views.py]: one field per line
'username': 'marcoantonio@gmail.com', 
'password': 'marcoantonio123456',

File: users/test_views.py (hunk 1 above)
Comment (marccolm) [inline users/test_views.py]: The api call returns a 204 code, which means that the email was sent, the email contains a link with a generated token and a uid, which must be sent as a POST to the confirm reset password url, which finally changes password, so I don't think there is a way to get the link

Comment (fernandolobato) [inline users/test_views.py]: Okey, thanks you.

**Pull request merged**